### PR TITLE
add model version attr to Model base object

### LINKED ISSUE #397: Make HydroMT aware of model version
### Kind of request

Adding new functionality

### Enhancement Description

t would be nice to make HydroMT aware of which model/software version it is working with. 

### Use case

For example, depending on software version, some additional input data might be needed and vice versa, or format of the readers/writers may be different. Tracking versions could also enable easily to update one model from a software version to another.

### Additional Context

Needs refinement but I guess we could add a model property for version and maybe work on dictionaries for which setup functions to use (and output to produce) and which readers/writers. Could also work within the functions with if/else statements but that would not make update of a model from one version to another easy.

### PR DESCRIPTION
**Pull request by savente93**
## Issue addressed
Fixes #397 

## Explanation
From our discussion today during the sprint planning I understand that this should be enough to close the issue. If anything else also needs to be done on this please let me know.

## Checklist
- [x] Updated tests or added new tests
- [x] Branch is up to date with `main`
- [x] Tests & pre-commit hooks pass
- [x] Updated documentation if needed
- [x] Updated changelog.rst if needed

## Additional Notes (optional)
Add any additional notes or information that may be helpful.

### CODE REVIEW

File: hydromt/models/model_api.py [hunk 1]
```diff
@@ -52,6 +52,8 @@ class Model(object, metaclass=ABCMeta):
     # TODO: change it back to setup_region and no res --> deprecation
     _CLI_ARGS = {"region": "setup_basemaps", "res": "setup_basemaps"}
     _TMP_DATA_DIR = None
+    # model version should be filled by the plugins
+    _MODEL_VERSION = None
```
Comment (DirkEilander) [inline hydromt/models/model_api.py:56]:
```suggestion
    # supported model version should be filled by the plugins
    # e.g. _MODEL_VERSION = ">=1.0, <1.1"
    _MODEL_VERSION = None
```

**Review by DirkEilander — CHANGES_REQUESTED** [review]
Just added a bit more context in the code in one suggestion.
And can you also add this to the changelog?

**Comment by savente93** [comment]
Good points thank you! Should be all good now :)

**Review by DirkEilander — APPROVED** [review]
Thanks!

**Pull request merged**